TopsortAnalytics/src/main/java/com/topsort/analytics/core/HttpClient.kt

```diff
@@ -33,7 +36,20 @@ class HttpClient (
         writeStream.flush()
         postConnection.outputStream.close()
 
-        val response = HttpResponse(connection.responseCode, connection.responseMessage)
+        val response = if (connection.responseCode in 200..299) {
```

Comment:
can we do `if (connection.responseCode not in 200..299) return …`

I find short-circuiting `return` easier to read since you avoid nesting code